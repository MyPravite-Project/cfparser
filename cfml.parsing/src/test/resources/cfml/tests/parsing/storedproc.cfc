function foo(){
    cfstoredproc(
         procedure="dbo.myTestProcedure",
         datasource=Variables.datasource
    ) {
         cfprocparam(value=myId, cfsqltype="cf_sql_integer");
         cfprocresult(resultset=1, name="myQuery");
    }
    Variables.myTest = [];
}