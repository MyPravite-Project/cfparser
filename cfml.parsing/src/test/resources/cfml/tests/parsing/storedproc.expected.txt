/*===TOKENS===*/
FUNCTION                 <function>
IDENTIFIER               <foo>
'('                      <(>
')'                      <)>
'{'                      <{>
IDENTIFIER               <cfstoredproc>
'('                      <(>
IDENTIFIER               <procedure>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <dbo.myTestProcedure>
CLOSE_STRING             <">
','                      <,>
IDENTIFIER               <datasource>
'='                      <=>
IDENTIFIER               <Variables>
'.'                      <.>
IDENTIFIER               <datasource>
')'                      <)>
'{'                      <{>
IDENTIFIER               <cfprocparam>
'('                      <(>
IDENTIFIER               <value>
'='                      <=>
IDENTIFIER               <myId>
','                      <,>
IDENTIFIER               <cfsqltype>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <cf_sql_integer>
CLOSE_STRING             <">
')'                      <)>
';'                      <;>
IDENTIFIER               <cfprocresult>
'('                      <(>
IDENTIFIER               <resultset>
'='                      <=>
INTEGER_LITERAL          <1>
','                      <,>
IDENTIFIER               <name>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <myQuery>
CLOSE_STRING             <">
')'                      <)>
';'                      <;>
'}'                      <}>
IDENTIFIER               <Variables>
'.'                      <.>
IDENTIFIER               <myTest>
'='                      <=>
'['                      <[>
']'                      <]>
';'                      <;>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (element
    (functionDeclaration
      function
      (identifier foo)
      (
      parameterList
      )
      (compoundStatement
        {
        (statement
          (startExpression
            (compareExpression
              (baseExpression
                (unaryExpression
                  (memberExpression
                    (functionCall
                      (identifier cfstoredproc)
                      (
                      (argumentList
                        (argument
                          (argumentName (identifier procedure))
                          =
                          (startExpression
                            (compareExpression
                              (baseExpression
                                (unaryExpression
                                  (primaryExpression
                                    (literalExpression (stringLiteral " (stringLiteralPart dbo.myTestProcedure) "))
                                  )
                                )
                              )
                            )
                          )
                        )
                        ,
                        (argument
                          (argumentName (identifier datasource))
                          =
                          (startExpression
                            (compareExpression
                              (baseExpression
                                (unaryExpression
                                  (memberExpression (identifier Variables) . (identifier datasource))
                                )
                              )
                            )
                          )
                        )
                      )
                      )
                      (compoundStatement
                        {
                        (statement
                          (startExpression
                            (compareExpression
                              (baseExpression
                                (unaryExpression
                                  (memberExpression
                                    (functionCall
                                      (identifier cfprocparam)
                                      (
                                      (argumentList
                                        (argument
                                          (argumentName (identifier value))
                                          =
                                          (startExpression
                                            (compareExpression
                                              (baseExpression (unaryExpression (memberExpression (identifier myId))))
                                            )
                                          )
                                        )
                                        ,
                                        (argument
                                          (argumentName (identifier cfsqltype))
                                          =
                                          (startExpression
                                            (compareExpression
                                              (baseExpression
                                                (unaryExpression
                                                  (primaryExpression
                                                    (literalExpression (stringLiteral " (stringLiteralPart cf_sql_integer) "))
                                                  )
                                                )
                                              )
                                            )
                                          )
                                        )
                                      )
                                      )
                                    )
                                  )
                                )
                              )
                            )
                          )
                          ;
                        )
                        (statement
                          (startExpression
                            (compareExpression
                              (baseExpression
                                (unaryExpression
                                  (memberExpression
                                    (functionCall
                                      (identifier cfprocresult)
                                      (
                                      (argumentList
                                        (argument
                                          (argumentName (identifier resultset))
                                          =
                                          (startExpression
                                            (compareExpression
                                              (baseExpression (unaryExpression (primaryExpression (literalExpression 1))))
                                            )
                                          )
                                        )
                                        ,
                                        (argument
                                          (argumentName (identifier name))
                                          =
                                          (startExpression
                                            (compareExpression
                                              (baseExpression
                                                (unaryExpression
                                                  (primaryExpression
                                                    (literalExpression (stringLiteral " (stringLiteralPart myQuery) "))
                                                  )
                                                )
                                              )
                                            )
                                          )
                                        )
                                      )
                                      )
                                    )
                                  )
                                )
                              )
                            )
                          )
                          ;
                        )
                        }
                      )
                    )
                  )
                )
              )
            )
          )
        )
        (statement
          (assignmentExpression
            (startExpression
              (compareExpression
                (baseExpression
                  (unaryExpression
                    (memberExpression (identifier Variables) . (identifier myTest))
                  )
                )
              )
            )
            =
            (startExpression
              (compareExpression
                (baseExpression (unaryExpression (primaryExpression (implicitArray [ ]))))
              )
            )
          )
          ;
        )
        }
      )
    )
  )
)
/*======*/
/*===DECOMPILE===*/
public function foo()   {
cfstoredproc(procedure = 'dbo.myTestProcedure', datasource = Variables.datasource)      {
cfprocparam(value = myId, cfsqltype = 'cf_sql_integer');
cfprocresult(resultset = 1, name = 'myQuery');

      };
Variables.myTest = [];

  }
/*======*/